resilience4j-framework-common/src/main/java/io/github/resilience4j/common/bulkhead/configuration/ThreadPoolBulkheadConfigurationProperties.java

```diff
@@ -54,34 +57,45 @@ public InstanceProperties getBackendProperties(String backend) {
 
     // Thread pool bulkhead section
     public ThreadPoolBulkheadConfig createThreadPoolBulkheadConfig(String backend) {
-        return createThreadPoolBulkheadConfig(getBackendProperties(backend));
+        return createThreadPoolBulkheadConfig(getBackendProperties(backend),backend, Optional.empty());
+    }
+
+    public ThreadPoolBulkheadConfig createThreadPoolBulkheadConfig(String backend,
+        Optional<Customizer<ThreadPoolBulkheadConfig.Builder>> customizer) {
+        return createThreadPoolBulkheadConfig(getBackendProperties(backend),backend, customizer);
     }
 
     public ThreadPoolBulkheadConfig createThreadPoolBulkheadConfig(
-        InstanceProperties instanceProperties) {
+        InstanceProperties instanceProperties, String backendName,
+        Optional<Customizer<ThreadPoolBulkheadConfig.Builder>> customizer) {
```

Comment:
Minor remark , can we use https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/lang/Nullable.html for the optional spring dependency injection as using Optional as method parameter type is really bad practice .

Reply:
DI is not happening here, its a helper method which is public because it is called by other configuration. But I removed `Optional` from the Customizer beans which is getting DI. See latest commit.

Reply:
yes that i what i mentioned in my comment for AbstractBulkheadConfigurationOnMissingBean class ;)  , and still u do not need Optional in the helper methods as a method parameters , just resilience4j @Nullable which indicate it can be be null and do ur null check inside the last method itself or ofNullable whatever u like

Reply:
I will make this change!